File: frontend/src/types/kite.ts
export interface KiteLastSync {
  synced_at: string
  status: string
  holdings_count: number | null
  positions_count: number | null
  error_message: string | null
}

export interface KiteStatus {
  configured: boolean
  api_key: string | null
  token_valid: boolean
  token_expiry: string | null
  last_sync: KiteLastSync | null
  login_url: string | null
}

export interface KiteConfig {
  configured: boolean
  api_key?: string
  has_secret?: boolean
  token_valid?: boolean
  token_expiry?: string | null
}

export interface KiteSyncResult {
  synced_at: string
  status: string
  holdings_count: number
  positions_count: number
  error_message: string | null
}

File: app/schemas/kite.py
from typing import Optional
from pydantic import BaseModel


class KiteLastSync(BaseModel):
    synced_at: str
    status: str
    holdings_count: Optional[int] = None
    positions_count: Optional[int] = None
    error_message: Optional[str] = None


class KiteStatus(BaseModel):
    configured: bool
    api_key: Optional[str] = None
    token_valid: bool
    token_expiry: Optional[str] = None
    last_sync: Optional[KiteLastSync] = None
    login_url: Optional[str] = None


class KiteSyncResult(BaseModel):
    synced_at: str
    status: str
    holdings_count: int
    positions_count: int
    error_message: Optional[str] = None
